File: algorithm/codility/Lesson6-Sorting/Distinct/java/Distinct.java

import java.util.Arrays;

public class Distinct {
    public int solution(int[] A){
        System.out.println(Arrays.toString(A));
        Arrays.sort(A);
        int count = 0;

        System.out.println(Arrays.toString(A));
        int N = A.length;

        if (N == 0) {
            return 0;
        }
        for (int i = 1; i < N; i++) {
            if (A[i]-1 != A[i]) {
                count += 1;
                
            }
        }
        return count;
        
    }


    public static void main(String[] args) {
    
        int[] A = {2,1,1,2,3,1};
        Distinct dis = new Distinct();
        int count = dis.solution(A);
        System.out.println(count);
        }
}

File: algorithm/codility/Lesson6-Sorting/Distinct/python/Distinct.py
"""

"""
def solution(A):
    A.sort()
    distict = 0
    # [1,1,1,2,2,3]
    N = len(A)

    if N == 0: 
        return 0

    count = 1
    for i in range(1, N):
        if A[i-1] != A[i]:
            count += 1
    return count

if __name__ == "__main__":

    A = [2,1,1,2,3,1]
    print(solution(A))


